File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.BusinessLayer/Reserva.cs
using LogisticStorage.DataLayer;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.BusinessLayer
{
    public class Reserva
    {
        public static List<ReservaEntity> GetMercaderiaMainItems()
        {
            ReservaDB DB = new ReservaDB();
            return DB.GetMercaderiaMainItems();
        }
        public static Int32 ReservarMercaderia(ReservaEntity Ent)
        {
            ReservaDB DB = new ReservaDB();
            DB.ReservarMercaderia(Ent);

            return Ent.MercaderiaId;
        }
        public static List<ReservaEntity> ReservaPedido(Int32 MercaderiaId)
        {
            ReservaDB DB = new ReservaDB();
            return DB.ReservaPedido(MercaderiaId);
        }
        public static List<ReservaEntity> ReservaResumen(Int32 MercaderiaId)
        {
            ReservaDB DB = new ReservaDB();
            return DB.ReservaResumen(MercaderiaId);
        }
    }
}

File: ProyectoMysql/server/BusinessLayer/Reserva.py
from DataLayer.ReservaDB import ReservaDB
from EntityLayer.ReservaEntity import ReservaMercaderiaOPModel
from Utilidades.Conexion.configMysql import StartTransaction, EndTransaction, Restore


class Reserva:

    def GetMercaderiaMainItems():
        try:
            data =  ReservaDB.GetMercaderiaMainItems()
            return data
        except Exception as e:
            print(e)    

    def ReservarMercaderia(Ent: ReservaMercaderiaOPModel):
        try:
            StartTransaction()
            data = ReservaDB.ReservarMercaderia(Ent)
            EndTransaction()
            return data
        except Exception as e:
            Restore()
            print(e)

    def ReservaPedido(MercaderiaId : int):
        try:
            data =  ReservaDB.ReservaPedido(MercaderiaId)
            return data
        except Exception as e:
            print(e)    
    
    def ReservaResumen(MercaderiaId : int):
        try:
            data =  ReservaDB.ReservaResumen(MercaderiaId)
            return data
        except Exception as e:
            print(e)    

    def ObtenerReservaOPItem(OrdenPedidoId : int):
        try:
            data =  ReservaDB.ObtenerReservaOPItem(OrdenPedidoId)
            return data
        except Exception as e:
            print(e)    